pkg/alertmanager/multitenant.go

```diff
@@ -663,6 +671,23 @@ func (am *MultitenantAlertmanager) loadAndSyncConfigs(ctx context.Context, syncR
 	return nil
 }
 
+func (am *MultitenantAlertmanager) waitInitialStateSync(ctx context.Context) error {
+	am.alertmanagersMtx.Lock()
```

Comment:
Since this method runs as part of initialization in Starting state, I would not expect to see locking or protection against modifications here. If that is removed, method is simplified, and can perhaps be moved to `starting` method?